server/listeners/transform-event.js

```diff
@@ -28,6 +28,32 @@ const getReferences = (payload, messageTypes) => {
   return [];
 };
 
+const getBy = (payload) => { //eslint-disable-line
+  let by = [];
+  if (_.isArray(payload.by)) {
+    for (let item of payload.by) {
+      if (item.endpoint && item.endpoint.id) {
+        by.push({ endpoint: { id: item.endpoint.id, token: item.endpoint.token } });
+      }
+      if (item.user && item.user.id) {
```

Comment:
The ID will be the user's email address here, correct? Will there ever be a situation where we might have a token and not the email address? Also, in looking at the AuthContext object, what is the best way to grab the email address?

looked like User._claims would have a key of '#/sys/em' ?